### ISSUE TITLE
2021-12-25

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#694]
271P30, HW3; (2021-12-15)
272P31, HW3(English version); (2021-12-15)
273P32, HW4; (2021-12-16)
274P33, HW4(English version); (2021-12-16)
275P34, transformer上; seq2seq; output length is determined by model; speech recognition; machine translation; 硬train一发; text-to-speech; seq2seq for chatbot; NLP == QA, QA == seq2seq; seq2seq for multi-label classification; seq2seq for object detection; input -> encoder -> decoder -> output; encoder is vec2vec; block = self-attention + FC; 残差网络, 输入+输出; layer norm; Add&Norm == residual + layer normalization; 原有block可以换一下顺序; (2021-12-16)
276P35, transformer下; decoder; auto-regressive(由自己的过去推出自己的未来); begin token; 除了中间部分, encoder和decoder是一样的; masked self-attention, 只能考虑已有的(过去的/左边的)资讯;  decoder怎样知道输出的长度? 推文接龙; 需要加上end token; non-autoregressive decoder(NAT); (1) another predictor for output length; (2) ignore after END token; Advantage: parallel; usually worse than AT; encoder-decoder; cross-attention; training; teacher forcing: using ground truth as input; copy mechanism; summarization; 可能会犯低级错误; guided attention; beam search; decoder needs randomness when generating sequence; 不知道怎样优化, 就用强化学习; scheduled sampling, training set不仅有正确的, 还有错误的; (2021-12-16)
277P36, non-autoregressive sequence generation; RNN is auto-regressive; Vanilla NAT; fertility; sequence-level knowledge distillation; noisy parallel decoding; evolution of NAT; NAT with iterative refinement; mask-predict; insertion transformer; multiple target words to predict? 
278Download Xshell again; uninstall old version; install new version xshell; done; install new version xftp; done; (2021-12-16)
279SSFC bug fix: Report error of phrase is not found; Not found phrase ; 只要在Not found phrase更新时报错就可以; done; (2021-12-17)
